runner/bsp/worker/src/mill/bsp/worker/MillBuildServer.scala

```diff
@@ -54,24 +57,28 @@ private class MillBuildServer(
   // Set when a session is completed, either due to reload or shutdown
   private[worker] var sessionResult: Option[BspServerResult] = None
 
+  private val requestCount = new AtomicInteger
+
   def initialized = sessionInfo != null
 
   def updateEvaluator(evaluatorsOpt: Option[Seq[EvaluatorApi]]): Unit = {
-    debug(s"Updating Evaluator: $evaluatorsOpt")
+    baseLogger.debug(s"Updating Evaluator: $evaluatorsOpt")
     if (bspEvaluators.isCompleted) bspEvaluators = Promise[BspEvaluators]() // replace the promise
     evaluatorsOpt.foreach { evaluators =>
-      bspEvaluators.success(new BspEvaluators(topLevelProjectRoot, evaluators, s => debug(s())))
+      bspEvaluators.success(new BspEvaluators(
+        topLevelProjectRoot,
+        evaluators,
+        s => baseLogger.debug(s())
+      ))
     }
   }
 
-  def print(msg: String): Unit = logStream.println(msg)
-  def debug(msg: => String): Unit = if (debugMessages) logStream.println("[debug] " + msg)
-
   def onConnectWithClient(buildClient: BuildClient): Unit = client = buildClient
 
   override def buildInitialize(request: InitializeBuildParams)
-      : CompletableFuture[InitializeBuildResult] =
-    handlerRaw(checkInitialized = false) {
+      : CompletableFuture[InitializeBuildResult] = {
+    val logger = createLogger()
```

Comment:
Do we really need to call `createLogger()` and pass it in to every `handlerRaw`? Could we instead make `handlerRaw` take the `sourcecode.Enclosing` and create the logger internally?

Reply:
Should be fixed